[JIT] IRParser: store list attributes as generic ivalue lists.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43785 [JIT] IRParser: store list attributes as generic ivalue lists.**

Differential Revision: [D23400565](https://our.internmc.facebook.com/intern/diff/D23400565)

Commit:
[JIT] IRParser: store list attributes as generic ivalue lists.

[ghstack-poisoned]

eellison reviewed (approved):
Nice, lgtm! There may be some nodes out there that use `is` and stuff, but those are usually pass specific and not used in IR parsing so we should land this.

Commit:
Update on "[JIT] IRParser: store list attributes as generic ivalue lists."


Differential Revision: [D23400565](https://our.internmc.facebook.com/intern/diff/D23400565)

[ghstack-poisoned]
